easy_format.ml

```diff
@@ -57,12 +65,14 @@ let list = {
 }
 
 type label_param = {
+  label_break: label_break;
   space_after_label : bool;
   indent_after_label : int;
   label_style : style_name option;
 }
 
 let label = {
+  label_break = `Auto;
```

Comment:
Ideally users would only use the record update syntax, e.g. `{ default_from_library with label_break = `Always }`, which tolerates the addition of future fields. Providing functions with optional arguments makes it a little more obvious I guess but doesn't let the user inherit and override some fields from other sources than the default:
```ocaml
(* provided by library *)
val create_config : ?a:a -> ?b:b -> unit -> config

(* user code *)
let my_default_config =  create_config ~a:`Whatever ()
let special_config = ???
```
In order to create `special_config`, currently we'd use the `with` syntax. If we only provided labeled functions, we'd need a higher-order function that does the same. It would have the following type:
```ocaml
(* library would provide this *)
val config_class : ?a:a -> ?b:b -> unit -> (?a:a -> ?b:b -> unit -> config)

(* user code *)
let create_config = config_class ~a:`Whatever ()
let my_special_config = create_config ~b:`Bananas ()
```
The user code will remain compatible when future options are added as long as `create_config` is left without a type annotation that would lock the number and name of the arguments.